server/src/models/user.js

```diff
@@ -0,0 +1,79 @@
+const mongoose = require("mongoose");
+const bcrypt = require("bcrypt");
+
+const userSchema = new mongoose.Schema({
+    firstName: {
+        type: String,
+        required: true,
+        trim: true,
+        min: 3,
+        max: 20
+    },
+    lastName: {
+        type: String,
+        required: true,
+        trim: true,
+        min: 3,
+        max: 20
+    },
+    userName: {
+        type: String,
+        required: true,
+        trim: true,
+        unique: true,
+        index: true,
+        lowercase: true
+
+    },
+    email: {
+        type: String,
+        required: true,
+        trim: true,
+        unique: true,
+        lowercase: true
+    },
+    hash_password: {
+        type: String,
+        required: true
+    },
+    role: {
+        type: String,
+        enum: ['user'],
```

Comment:
If there is only one role then why is there a need of role table at all?

Reply:
in the future, if we would like to introduce more features and for that, we need admin then?

Reply:
@himanshujaidka Have a converation with the project admin and if it's something we intend to do then add admin role too.